test/general/openpgp.js

```diff
@@ -1821,6 +1846,24 @@ aOU=
       })).to.be.rejectedWith(/No signing keys provided/);
     });
 
+    it('Signing with key which uses sha3 should generate a valid sha3 signature', async function() {
+      const privKey = await openpgp.readKey({ armoredKey: priv_key_sha3 });
+      const pubKey = privKey.toPublic();
+      const text = 'Hello, world.';
+      const message = await openpgp.createCleartextMessage({ text });
+      const config = { minRSABits: 1024 };
+
+      const cleartextMessage = await openpgp.sign({ message, signingKeys: privKey, config, format: 'armored' });
+      const parsedArmored = await openpgp.readCleartextMessage({ cleartextMessage });
+      expect(parsedArmored.text).to.equal(text);
+      expect(parsedArmored.signature.packets.filterByTag(openpgp.enums.packet.signature)).to.have.length(1);
```

Comment:
Could you add a check that this filtered signature packet has `.hashAlgorithm` prop that equals `openpgp.enums.hash.sha3_x` (not sure what sha3 size it uses)?

Reply:
Yes! Thanks so much, I thought about this but I wasn't sure exactly how to check the alg in the signature. :)